flocker/node/agents/ebs.py

```diff
@@ -107,6 +115,68 @@ def _wait_for_volume(expected_volume,
             )
 
 
+def _check_blockdevice_size(device, size):
+    """
+    Helper function to check if the size of block device with given
+    suffix matches the input size.
+
+    :param unicode device: Name of the block device to check for size.
+    :param int size: Size, in SI metric bytes, of device we are interested in.
+
+    :returns True if a block device with given name has given size.
+        False otherwise.
+    """
+    device_name = b"/dev/" + device
+    command = [b"lsblk", b"-nb", b"-o", b"SIZE", device_name]
+    device_size = int(check_output(command).strip().decode("ascii"))
+
+    # OS reports device size in powers of 2, where has EBS
+    # talks powers of 10. Hence, convert before comparing.
+    if size/(1000**3) == device_size/(1024**3):
```

Comment:
Having seen your use of `lsblk` above and realizing that the existing test suite didn't catch the GB/GiB mismatch, we should probably have a couple new tests in the test suite generated by `make_iblockdevice_tests`.  One (most importantly) which creates a volume and then attaches it and uses `lsblk` to verify it has the right size.  And another similar test for resize (but since we're probably dropping resize from 1.0 that can be lower priority).  This would provide an outside confirmation that sizes were being handled correctly instead of relying on the object being tested to tell us.